cfdm/test/test_NetCDFArray.py

```diff
@@ -0,0 +1,79 @@
+import atexit
+import datetime
+import faulthandler
+import os
+import tempfile
+import unittest
+
+faulthandler.enable()  # to debug seg faults and timeouts
+
+import cfdm
+
+n_tmpfiles = 1
+tmpfiles = [
+    tempfile.mkstemp("_test_netCDF.nc", dir=os.getcwd())[1]
+    for i in range(n_tmpfiles)
+]
+(tmpfile,) = tmpfiles
+
+
+def _remove_tmpfiles():
+    """Remove temporary files created during tests."""
+    for f in tmpfiles:
+        try:
+            os.remove(f)
+        except OSError:
+            pass
+
+
+atexit.register(_remove_tmpfiles)
+
+
+class NetCDFTest(unittest.TestCase):
+    """Unit test for the NetCDF class."""
+
+    def setUp(self):
+        """Preparations called immediately before each test method."""
+        # Disable log messages to silence expected warnings
+        cfdm.log_level("DISABLE")
+        # Note: to enable all messages for given methods, lines or
+        # calls (those without a 'verbose' option to do the same)
+        # e.g. to debug them, wrap them (for methods, start-to-end
+        # internally) as follows:
+        #
+        # cfdm.log_level('DEBUG')
+        # < ... test code ... >
+        # cfdm.log_level('DISABLE')
+
+    def test_NetCDFArray_get_missing_values(self):
+        """Test NetCDFArray.get_missing_values."""
+        f = cfdm.example_field(0)
+
+        f.set_property("missing_value", -999)
+        f.set_property("_FillValue", -3)
+        f.set_property("valid_range", [-111, 222])
+        cfdm.write(f, tmpfile)
+
+        g = cfdm.read(tmpfile)[0]
+        print(g.data.source().get_missing_values())
```

Comment:
```suggestion
        g.data.source().get_missing_values()
```

so the test doesn't produce output which clogs the STDOUT report, unless you deliberately want to check the `__str__` representation?

Reply:
Ah - I think delete altogether. Good spot. https://github.com/NCAS-CMS/cfdm/pull/247/commits/41ad49c45273706c7d3e81d4c9ebbfc3a2bfbebc